### PR TITLE
Update Working with Dependencies -> Version Section Cleanup

### PR DESCRIPTION
**Pull request by lkasso**
### Details
This is a Documentation change ONLY.

### Context
Clean up existing pages in the [Declaring Versions](https://docs.gradle.org/current/userguide/single_versions.html) section of the Dependency Management docs.

### Review
[Link]()

### CODE REVIEW

**Comment by lkasso** [comment]
@bot-gradle test BD

File: platforms/documentation/docs/src/docs/userguide/dep-man/02-declaring-dependency-versions/dynamic_versions.adoc [hunk 1]
```diff
@@ -13,89 +13,91 @@
...
-This can be a requirement during development, or you may be developing a library that is designed to work with a range of dependency versions.
 You can easily depend on these constantly changing dependencies by using a _dynamic version_.
-A <<dynamic_versions.adoc#sub:declaring_dependency_with_dynamic_version,dynamic version>> can be either a version range (e.g. `2.+`) or it can be a placeholder for the latest version available e.g. `latest.integration`.
 
-Alternatively, the module you request can change over time even for the same version, a so-called <<dynamic_versions.adoc#sub:declaring_dependency_with_changing_version,changing version>>.
-An example of this type of _changing module_ is a Maven `SNAPSHOT` module, which always points at the latest artifact published.
-In other words, a standard Maven snapshot is a module that is continually evolving, it is a "changing module".
+A <<dynamic_versions.adoc#sub:declaring_dependency_with_dynamic_version,dynamic version>> can be either a version range (e.g., `2.+`) or a placeholder for the latest version available, e.g., `latest.integration`.
```
Comment (jvandort) [inline platforms/documentation/docs/src/docs/userguide/dep-man/02-declaring-dependency-versions/dynamic_versions.adoc:22]: It would be nice if we can specifically call out dependency locking here. 

I would say in general, if you are using dynamic versions, you should _almost always_ use dependency locking. Without dependency locking, dynamic versions lead to non-reproducible builds -- which should be avoided if at all possible. 

Would we be able to add a quick note/warning here saying you should almost always use dependency locking along with dynamic versions?

File: platforms/documentation/docs/src/docs/userguide/dep-man/02-declaring-dependency-versions/dynamic_versions.adoc [hunk 2]
```diff
@@ -13,89 +13,91 @@
...
 [[sub:declaring_dependency_with_dynamic_version]]
 == Declaring a dynamic version
 
 Projects might adopt a more aggressive approach for consuming dependencies to modules.
-For example you might want to always integrate the latest version of a dependency to consume cutting edge features at any given time.
+For example, you might want to integrate the latest version of a dependency to consume cutting-edge features at any given time.
 A _dynamic version_ allows for resolving the latest version or the latest version of a version range for a given module.
 
-CAUTION: Using dynamic versions in a build bears the risk of potentially breaking it. As soon as a new version of the dependency is released that contains an incompatible API change your source code might stop compiling.
+CAUTION: Using dynamic versions in a build can break it. As soon as a new version of the dependency that contains an incompatible API change is released, your source code might stop compiling.
```
Comment (jvandort) [inline platforms/documentation/docs/src/docs/userguide/dep-man/02-declaring-dependency-versions/dynamic_versions.adoc:37]: This is what I was talking about. Because of this reason, you want to use dependency locking with dynamic versions.

File: platforms/documentation/docs/src/docs/userguide/dep-man/02-declaring-dependency-versions/single_versions.adoc [hunk 3]
```diff
@@ -15,28 +15,62 @@
...
-For example versions like `2.0-dev1` or `2.0-SNAPSHOT` are no longer included in the range.
-* A _prefix_ version range: e.g. `1.\+`, `1.3.+`
-** Only versions exactly matching the portion before the `+` are included.
-** The range `+` on its own will include any version.
-* A `latest-status` version: e.g. `latest.integration`, `latest.release`
-** Will match the highest versioned module with the specified status. See link:{javadocPath}/org/gradle/api/artifacts/ComponentMetadata.html#getStatus--[ComponentMetadata.getStatus()].
-* A Maven `SNAPSHOT` version identifier: e.g. `1.0-SNAPSHOT`, `1.4.9-beta1-SNAPSHOT`
+[cols="1,1,~"]
+|===
+|Version |Example |Note
```
Comment (jvandort) [inline platforms/documentation/docs/src/docs/userguide/dep-man/02-declaring-dependency-versions/single_versions.adoc:38]: Looking at the generated table, this `Version` column looks very squished. Is there a way to make it a bit wider?

**Review by jvandort — APPROVED** [review]
